docs/language/ql-training/slide-snippets/abstract-syntax-tree.rst

```diff
@@ -47,23 +47,23 @@ The basic representation of an analyzed program is an *abstract syntax tree (AST
 Database representations of ASTs
 ================================
 
-AST nodes and other program elements are encoded in the database as *entity values*. Entities are implemented as integers, but in QL they are opaque - all one can do with them is to check their equality.
+AST nodes and other program elements are encoded in the database as *entity values*. Entities are implemented as integers, but in QL they are opaque---all one can do with them is to check their equality.
 
 Each entity belongs to an entity type. Entity types have names starting with “@” and are defined in the database schema (not in QL).
 
 Properties of AST nodes and their relationships to each other are encoded by database relations, which are predicates defined in the database (not in QL).
```

Comment:
(I'm not sure about the usage of QL here, but it's probably not worth worrying about for now!)

Reply:
Yeah, I left this as-is because I couldn't see anything specific to code 🤷‍♂